src/Facebook/InstantArticles/Elements/Caption.php

```diff
@@ -345,62 +317,60 @@ public function getPosition()
      * @see Caption::VERTICAL_BOTTOM
      * @see Caption::VERTICAL_CENTER
      */
-    public function getVerticalAlignment()
+    public function getVerticalAlignment(): string
     {
         return $this->verticalAlignment;
     }
 
     /**
-     * Structure and create the full ArticleImage in a XML format DOMElement.
+     * Structure and create the full ArticleImage in a XML format DOMNode.
      *
      * @param \DOMDocument $document where this element will be appended. Optional
-     * @return \DOMElement
+     * @return \DOMNode
      */
-    public function toDOMElement($document = null)
+    public function toDOMElement(\DOMDocument $document): \DOMNode
     {
-        if (!$document) {
-            $document = new \DOMDocument();
-        }
-
         if (!$this->isValid()) {
             return $this->emptyElement($document);
         }
 
         $element = $document->createElement('figcaption');
 
         // title markup REQUIRED
-        if ($this->title && (!$this->subTitle && !$this->credit)) {
-            $element->appendChild($this->title->textToDOMDocumentFragment($document));
-        } elseif ($this->title) {
-            $element->appendChild($this->title->toDOMElement($document));
+        if ($this->title !== null) {
+            if (!$this->subTitle && !$this->credit) {
```

Comment:
Lets swap the if/else conditions and add some comments.

```php
if ($this->subTitle != null || $this->credit != null) {
    // Use the <h1> node if a sub title or credit is present
} else {
    // Use a simple loose text node if the caption is by itself
}
```

Reply:
Good catch!
Fixing